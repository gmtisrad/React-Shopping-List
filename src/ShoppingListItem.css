.list-item {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    background-color: white;
    border: 1px solid black;
    color: slategray;
    padding: 2px 0 2px 10px;
    align-items: center;
    margin: 4px 0
}

.item-buttons {
    display: flex;
    flex-direction: column;
    text-decoration: none;
}

.item-buttons button {
    padding: 10px 20px;
}

.is-checked  p{
    text-decoration: line-through;
}