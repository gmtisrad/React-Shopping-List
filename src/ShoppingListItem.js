import React from 'react';
import './ShoppingListItem.css';

class ShoppingListItem extends React.Component {
    static defaultProps = {
        name: '',
        checked: null,
    }
    render() {
        return(
            <div className='list-item'>
                <p
                    style={{textDecoration: this.props.checked ? 'line-through' : null}}
                >
                    {this.props.name}
                </p>
                <div className='item-buttons'>
                    <button 
                        className='delete'
                        onClick={() => this.props.onDeleteItem(this.props.name)}
                    >
                    Delete
                    </button>
                    <button 
                        className='check'
                        onClick={() => this.props.onCheckItem(this.props.name)}
                    >
                    Check
                    </button>
                </div>
            </div>
        )
    }
}

export default ShoppingListItem;