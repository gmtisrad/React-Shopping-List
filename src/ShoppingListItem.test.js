import ReactDOM from 'react-dom';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShoppingListItem from './ShoppingListItem.js';

describe ('ShoppingListeItem tests', () => {
    it('renders without crash', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ShoppingListItem />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI without props as expected', () => {
        const wrapper = shallow(<ShoppingListItem />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})