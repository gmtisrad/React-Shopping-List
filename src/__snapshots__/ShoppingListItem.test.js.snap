// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingListeItem tests renders UI without props as expected 1`] = `
<div
  className="list-item"
>
  <p
    style={
      Object {
        "textDecoration": null,
      }
    }
  />
  <div
    className="item-buttons"
  >
    <button
      className="delete"
    >
      Delete
    </button>
    <button
      className="check"
    >
      Check
    </button>
  </div>
</div>
`;
